test/sharness/Makefile

```diff
@@ -1,57 +1,7 @@
-# Run tests
-#
-# Copyright (c) 2014 Christian Couder
-# MIT Licensed; see the LICENSE file in this repository.
-#
+SH := $(wildcard *.sh)
 
-# NOTE: Run with TEST_VERBOSE=1 for verbose sharness tests.
+.DEFAULT $(SH): ALWAYS
+	$(MAKE) -C ../.. test/sharness/$@
 
-T = $(sort $(wildcard t[0-9][0-9][0-9][0-9]-*.sh))
-BINS = bin/random bin/multihash bin/ipfs bin/pollEndpoint \
-	   bin/iptb bin/go-sleep bin/random-files bin/go-timeout bin/hang-fds
-SHARNESS = lib/sharness/sharness.sh
-IPFS_ROOT = ../..
-
-# User might want to override those on the command line
-GOFLAGS =
-
-all: aggregate
-
-clean: clean-test-results
-	@echo "*** $@ ***"
-	rm -rf $(BINS)
-
-clean-test-results:
-	@echo "*** $@ ***"
-	rm -rf test-results
-
-$(T): clean-test-results deps
-	@echo "*** $@ ***"
-	./$@
-
-aggregate: clean-test-results $(T)
-	@echo "*** $@ ***"
-	lib/test-aggregate-results.sh
-
-global-deps:
-	$(MAKE) -C ../.. deps
-
-deps: global-deps $(SHARNESS) $(BINS) curl
-
-$(SHARNESS): FORCE
-	@echo "*** checking $@ ***"
-	lib/install-sharness.sh
-
-bin/%: FORCE
-	$(MAKE) -C .. GOFLAGS=$(GOFLAGS) $@
-
-race:
-	$(MAKE) GOFLAGS=-race all
-
-.PHONY: all clean $(T) aggregate FORCE
-
-# will fail if curl is not installed.
-# TODO: get rid of this and install curl with git or ipfs.
-install_curl_pls = "curl is required to run tests. please install it"
-curl:
-	@which curl >/dev/null || (echo "$(install_curl_pls)" && false)
+ALWAYS: 
+.PHONY: ALWAYS
```

Comment:
When I try to just run make in `test/sharness` I get this
```
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
Makefile:3: target 'trash' given more than once in the same rule
...
rm -rf test/sharness/test-results
*** test/sharness/t0500-issues-and-regressions-offline.sh ***
...
```
That is a get a bunch of warnings and only a single test runs.

Also it seams now every time I run a single test it seams to try and rebuild all the deps:
```
$ time make  t0210-tar.sh
...
make -C ../.. test/sharness/t0210-tar.sh
make[1]: Entering directory '/aux/home/gocode2/src/github.com/ipfs/go-ipfs'
bin/check_go_version 1.7
gx install --global >/dev/null 2>&1
go build -i  -o "test/bin/random" "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-random/random"
go build -i  -o "test/bin/multihash" "gx/ipfs/QmYDds3421prZgqKbLpEK7T9Aa2eVdQ7o3YarX1LVLdP2J/go-multihash/multihash"
go build -i  -o "test/bin/pollEndpoint" "github.com/ipfs/go-ipfs/thirdparty/pollEndpoint"
go build -i  -o "test/bin/iptb" "gx/ipfs/QmSXCTgRb3JhrV5ALbWocbNGD8Tc3HVHczFYvYKrSQAau3/iptb"
go build -i  -o "test/bin/go-sleep" "github.com/ipfs/go-ipfs/test/dependencies/go-sleep"
go build -i  -o "test/bin/random-files" "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-random-files/random-files"
go build -i  -o "test/bin/go-timeout" "github.com/ipfs/go-ipfs/test/dependencies/go-timeout"
go build -i  -o "test/bin/hang-fds" "gx/ipfs/QmPRYAPKG9MhfJLrDT659SGhBsY7V7zHq84cngi8EHudHQ/hang-fds"
go build -i  -o "cmd/ipfs/ipfs" "github.com/ipfs/go-ipfs/cmd/ipfs"
rm -rf test/sharness/test-results
*** test/sharness/t0210-tar.sh ***
...
real    0m10.183s
user    0m9.132s
sys     0m1.208s
```